test/lit_tests/jit_call.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: enzymexlamlir-opt %s --pass-pipeline="builtin.module(lower-jit{jit=false backend=cpu},enzyme-hlo-opt)" | FileCheck %s
+
+module {
+  llvm.func internal unnamed_addr fastcc @throw() attributes {dso_local, no_inline, sym_visibility = "private"} {
+    llvm.unreachable
+  }
+
+  // CHECK-LABEL: func.func @main(%arg0: tensor<4xf32>) -> tensor<4xf32> {
+  func.func @main(%arg0: tensor<4xf32>) -> tensor<4xf32> {
+    // CHECK: stablehlo.custom_call @enzymexla_compile_cpu() {api_version = 3 : i32, backend_config = "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00", has_side_effect = true} : () -> ()
```

Comment:
shouldn't the `backend_config` remain the same and get passed to the underlying func?

Reply:
this needs fixing, the backend config needs to be compiled as part of CompileCall and this backend config should just be the attribute (at least that is what I figured out from the discussion above)